src/main/flight/mixer.c

```diff
@@ -502,16 +504,20 @@ static void applyMixerAdjustmentLinear(float *motorMix, const bool airmodeEnable
     throttle = constrainf(throttle, -minMotor, 1.0f - maxMotor);
 }
 
-static float calcEzLandLimit(float maxDeflection)
+static float calcEzLandLimit(float maxDeflection, float speed)
 {
     // calculate limit to where the mixer can raise the throttle based on RPY stick deflection
     // 0.0 = no increas allowed, 1.0 = 100% increase allowed
-    float ezLandLimit = 1.0f;
-    if (maxDeflection < mixerRuntime.ezLandingThreshold) { // roll, pitch and yaw sticks under threshold
-        ezLandLimit = maxDeflection / mixerRuntime.ezLandingThreshold; // normalised 0 - 1
-        ezLandLimit = fmaxf(ezLandLimit, mixerRuntime.ezLandingLimit); // stay above the minimum
-    }
-    return ezLandLimit;
+    const float deflectionLimit = fminf(1.0f, maxDeflection / mixerRuntime.ezLandingThreshold);
+
+    // calculate limit to where the mixer can raise the throttle based on speed
+    // TODO sanity checks like number of sats, dop, accuracy?
+    const float speedLimit = mixerRuntime.ezLandingSpeed > 0.0f ? fminf(1.0f, speed / mixerRuntime.ezLandingSpeed) : 0.0f;
+    DEBUG_SET(DEBUG_EZLANDING, 5, lrintf(speedLimit * 10000.0f));
```

Comment:
```suggestion
    DEBUG_SET(DEBUG_EZLANDING, 4, lrintf(speedLimit * 10000.0f));
```

Reply:
I have added the limit from the stick movement to EZLANDING debug 4 now